src/main/java/TaskManager.java

```diff
@@ -0,0 +1,191 @@
+
+            String taskType = tasks[i].getType();
+            String taskStatus = tasks[i].getStatusIcon();
+            String taskDescription = tasks[i].getDescription();
+
+            if (taskType.equals(TASK_TYPE_ICON_TODO)) {
+                System.out.printf("[%s][%s] %s%n", taskType, taskStatus, taskDescription);
+            } else if (taskType.equals(TASK_TYPE_ICON_DEADLINE)) {
+                String taskByTime = tasks[i].getByDateTime();
+                System.out.printf("[%s][%s] %s(by:%s)%n", taskType, taskStatus, taskDescription, taskByTime);
+            } else if (taskType.equals(TASK_TYPE_ICON_EVENT)) {
+                String taskAtTime = tasks[i].getStartAndEndTime();
+                System.out.printf("[%s][%s] %s(at:%s)%n", taskType, taskStatus, taskDescription, taskAtTime);
+            } else {
+                printGenericErrorMessage();
+            }
+        }
+    }
+
+    public void addNewTodo(String description) {
+        tasks[taskCount] = new ToDo(description);
+        taskCount++;
+    }
+
+    public void addNewDeadline(String description, String byDateTime) {
+        tasks[taskCount] = new Deadline(description, byDateTime);
+        taskCount++;
+    }
+
+    public void addNewEvent(String description, String startAndEndTime) {
+        tasks[taskCount] = new Event(description, startAndEndTime);
+        taskCount++;
+    }
+
+    public void printAddedTaskMessage(Task task) {
+
+        String taskType = task.getType();
+        String taskDescription = task.getDescription();
+
+        System.out.println("Got it. I've added this task:");
+        if (taskType.equals(TASK_TYPE_ICON_TODO)) {
+            System.out.printf("[%s][ ] %s%n", taskType, taskDescription);
+        } else if (taskType.equals(TASK_TYPE_ICON_DEADLINE)) {
+            String byTime = task.getByDateTime();
+            System.out.printf("[%s][ ] %s(by:%s)%n", taskType, taskDescription, byTime);
+        } else if (taskType.equals((TASK_TYPE_ICON_EVENT))) {
+            String atTime = task.getStartAndEndTime();
+            System.out.printf("[%s][ ] %s(at:%s)%n", taskType, taskDescription, atTime);
+        } else {
+            printGenericErrorMessage();
+        }
+
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+    }
+
+    public void markAsDone(Task task) {
+        if (task.getStatusIcon().equals(ICON_DONE)) {
+            printAlreadyDoneMessage();
+        } else {
+            task.setDone();
+            printMarkAsDoneMessage(task);
+        }
+    }
+
+    public void parseUserInput() {
+
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+            String line = in.nextLine();
+            if (line.equals(COMMAND_BYE)) {
+                break;
+            }
+            printLine();
+            String[] lineArgs = line.split(SEPARATOR_SPACE);
+            String command = lineArgs[0];
+            if (line.equals(COMMAND_LIST)) {
```

Comment:
Consider using switch statements to improve readability? Feels like nested if-else statements can be confusing sometimes